### PR TITLE
Fixes for USNG calculations that fail near the prime meridian and international date line

### PR DESCRIPTION
**Pull request by mverret**
This also adds unit tests to check for edge case issues as well as providing a means of regression testing.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "usng.js",
+  "version": "0.0.1",
+  "devDependencies": {
+    "chai": "^1.10.0",
+    "grunt": "latest",
```
Comment (pklinef) [inline package.json]: Is this safe?  What if it updated to a version of grunt that is not backwards compatible with the other dev dependencies?

File: tests/tests.js [hunk 2]
```diff
@@ -0,0 +1,693 @@
+var chai = require('chai');
+var usngs = require('../usng');
+var converter = new usngs.Converter();
+
```
Comment (pklinef) [inline tests/tests.js]: You should add tests for points and bounding boxes going both directions for every hemisphere too.

Comment (mverret) [inline tests/tests.js]: Added a test for a bounding box in North America, South America, Madagascar and Japan.  I also put tests in to handle the 100k, 10k and 1k resolutions for those new bounding boxes.
